src/main/java/games/strategy/debug/ConsoleHandler.java

```diff
@@ -0,0 +1,105 @@
+package games.strategy.debug;
+
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.logging.ErrorManager;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.LogRecord;
+import java.util.logging.SimpleFormatter;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * A {@link Handler} that publishes log records of level {@link Level#WARNING} or {@link Level#SEVERE} to
+ * {@code System.err} and log records of all other levels to {@code System.out}.
+ *
+ * <p>
+ * <strong>Configuration:</string> This handler does not currently support configuration through the {@link LogManager}.
+ * It always uses the following default configuration:
+ * </p>
+ *
+ * <ul>
+ * <li>Level: {@code Level.ALL}</li>
+ * <li>Filter: No {@code Filter}</li>
+ * <li>Formatter: {@code java.util.logging.SimpleFormatter}</li>
+ * <li>Encoding: default platform encoding</li>
+ * </ul>
+ */
+public final class ConsoleHandler extends Handler {
+  private final Supplier<PrintStream> errSupplier;
+  private final Supplier<PrintStream> outSupplier;
```

Comment:
I'm using `Supplier`s instead of direct references to the streams to avoid a potential race condition with the code in `ErrorConsole` that changes the system stdout and stderr streams.  If the `ConsoleHandler` instance happens to be created before the `ErrorConsole` is initialized (which is the way I've set it up in this PR), and it stores direct references to `System.out` and `System.err` at that time, it will bypass the new streams set up by `ErrorConsole`.  The implication of that is that no log messages will go to the `ErrorConsole` window (although they will still be sent to the underlying stdout and stderr streams and be visible, for example, in the terminal).  By using a `Supplier`, we get the current references to which these streams are pointing every time we write a log message.

I could have simply ordered things in such a way so that we would get the "correct" streams when using direct references.  However, I could hear the cries of a future TripleA maintainer cursing whoever required the `ErrorConsole` to be created before the logging system was initialized. :smile:  So, I figured a bit of indirection encapsulated in the implementation of one class was a better choice.